astroid/nodes/node_ng.py

```diff
@@ -309,13 +309,12 @@ def scope(self) -> nodes.LocalsDictNodeNG:
             raise ParentMissingError(target=self)
         return self.parent.scope()
 
-    def root(self) -> nodes.Module:
+    def root(self) -> nodes.Module | nodes.Unknown:
         """Return the root node of the syntax tree.
 
         :returns: The root node.
         """
         if not (parent := self.parent):
-            assert isinstance(self, nodes.Module)
```

Comment:
Shouldn't we update this `isinstance` check?

Reply:
I guess it was here to appease mypy... well mypy's green, but huh.. yeah.